### PR TITLE
Support monolithic deployment mode

### LINKED ISSUE #710: Support Tempo deployments in monolithic mode
The operator should support Tempo deployments in [monolithic mode](https://grafana.com/docs/tempo/latest/setup/deployment/#monolithic-mode).
For example local filesystem and in-memory storage is only available in monolithic deployments.

Design doc: https://docs.google.com/document/d/1fbBfXddfukj6Pfkr8rtRfUj522pIk5JKBzA_WyppPW0

```[tasklist]
### Tasks
- [x] In-memory storage: #722
- [x] PV storage: #722
- [x] Object storage: #763
- [x] Ingestion OTLP/gRPC: #722
- [x] Ingestion OTLP/HTTP: #722
- [x] Ingestion TLS: #756
- [x] Jaeger UI: #722
- [x] Jaeger UI Ingress+Route: #755
- [x] Multitenancy/Gateway: #816
- [x] Observability: ServiceMonitor, PrometheusRules, GrafanaDatasource: #793
- [x] Overlay config: #722
- [x] Management State: #722
- [x] Resources/Limits: #771
- [x] Operand status field: #787
- [x] Upgrade: #850
- [x] Tolerations (#782)
- [x] support custom service account: #836
- [x] TLS Profile: #862
- [ ] watch external resources (storage secret, T config map and secret)
```

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
Support Tempo monolithic deployment mode with a new `TempoMonolithic` CR.
Partially resolves #710.

- [X] In-memory storage
- [X] PV storage
- [X] Ingestion OTLP/gRPC
- [X] Ingestion OTLP/HTTP
- [X] Jaeger UI
- [X] Overlay config
- [X] Management State
- [x] Tests

### CODE REVIEW

File: .env
Comment (pavolloffay) [inline .env:1]: Why is this file needed? When are the tags updated?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+// MonolithicObservabilitySpec defines the observability settings of the Tempo deployment.
+type MonolithicObservabilitySpec struct {
+	// Metrics defines the metrics configuration of the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Metrics *MonolithicObservabilityMetricsSpec `json:"metrics,omitempty"`
+}
+
+// MonolithicObservabilityMetricsSpec defines the metrics settings of the Tempo deployment.
+type MonolithicObservabilityMetricsSpec struct {
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:201]: Can't we reuse smth from the microservice type? 

e.g. the whole observability spec https://github.com/grafana/tempo-operator/blob/cbe369b4983893cf50ce60201a1ed055f71bc344/apis/tempo/v1alpha1/tempostack_types.go#L141

File: apis/tempo/v1alpha1/tempomonolithic_webhook.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// SetupWebhookWithManager will setup the manager to manage the webhooks.
+func (r *TempoMonolithic) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type.
+func (r *TempoMonolithic) Default() {
+	log := ctrl.Log.WithName("tempomonolithic-webhook")
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_webhook.go:19]: how is this rendered in the logs? Isn't it too long?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+}
+
+// MonolithicTracesStorageBackend defines the backend storage for traces.
+//
+// +kubebuilder:validation:Enum=memory;pv
+type MonolithicTracesStorageBackend string
+
+const (
+	// MonolithicTracesStorageBackendMemory specifies a in-memory storage backend.
+	MonolithicTracesStorageBackendMemory MonolithicTracesStorageBackend = "memory"
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:75]: I think we should explain more what does the in-memory mean as there is no true in-memory storage in tempo.

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+	// MonolithicTracesStorageBackendPersistentVolume specifies a Persistent Volume storage backend.
+	MonolithicTracesStorageBackendPersistentVolume MonolithicTracesStorageBackend = "pv"
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:77]: will S3 be supported?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+	// +kubebuilder:validation:Required
+	Traces MonolithicTracesStorageSpec `json:"traces"`
+}
+
+// MonolithicTracesStorageSpec defines the traces storage for the Tempo deployment.
+type MonolithicTracesStorageSpec struct {
+	// Backend defines the backend for storing traces
+	//
+	// +kubebuilder:validation:Required
+	Backend MonolithicTracesStorageBackend `json:"backend"`
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:55]: What is the default?

File: docs/operator/api.md
Comment (pavolloffay) [inline docs/operator/api.md:1]: why there are so many changes in this file?

File: docs/spec/tempomonolithic.yaml
Comment (pavolloffay) [inline docs/spec/tempomonolithic.yaml:1]: How is this file generated? Is the tool for generating part of the repo tooling and CI check for changes? 

I would prefer to introduce it in a separate PR and align the microservices one with this as well

Comment (andreasgerstmayr) [reply docs/spec/tempomonolithic.yaml:1]: I've updated #711 to include the generation in the `api-docs` Makefile target

File: .env
Comment (andreasgerstmayr) [reply .env:1]: It's currently only used for the `run` Makefile target. I'm also not happy with the duplication of `.env` and `config/manager/manager.yaml`. Maybe we should remove the values from `manager.yaml` and use the ones from `.env` for it?

Or add these default image locations to the top of the Makefile?

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 4 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:77]: Yes, in a follow-up PR.

File: docs/operator/api.md
Comment (andreasgerstmayr) [reply docs/operator/api.md:1]: afaics just a bad diff algorithm, it detects insertions as edits.
This file contains a alphabetic-sorted list of all used structs in our CRDs, and the monolithic spec adds a couple more structs, inserting them between the other already existing ones. And the diff algorithm thinks it was an edit rather than an insert.

File: apis/tempo/v1alpha1/tempomonolithic_webhook.go (hunk 2 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_webhook.go:19]: It'll print this line:
```
{"level":"debug","ts":"2024-01-15T19:07:08.746058013+01:00","logger":"tempomonolithic-webhook","msg":"running defaulter webhook","name":"sample"}
```
if debug logs are enabled (`go run ./main.go --zap-log-level=debug start`).

Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_webhook.go:19]: But as we'll set the defaults in the reconcile loop now, I'm removing this log statement.

I'll keep it there for the validating webhook, which is still in use.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 3 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:76]: I've updated the comment

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 5 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:56]: `memory`, I've updated the comment and marker.

File: .env
Comment (andreasgerstmayr) [reply .env:1]: I've extracted this change to #746

File: .chloggen/monolithic_mode.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Support monolithic deployment mode
+
+# One or more tracking issues related to the change
+issues: [710]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (pavolloffay) [inline .chloggen/monolithic_mode.yaml:16]: Can we have more details here? This is important and cool new feature. It would be good to mention the new type and memory storage.

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+type MonolithicStorageSpec struct {
+	// Traces defines the backend storage configuration for traces
+	//
+	// +kubebuilder:validation:Required
+	Traces MonolithicTracesStorageSpec `json:"traces"`
+}
+
+// MonolithicTracesStorageSpec defines the traces storage for the Tempo deployment.
+type MonolithicTracesStorageSpec struct {
+	// Backend defines the backend for storing traces. Default: memory
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:52]: I am a bit concerned about using the in-memory as default. 

The upstream uses PV as default and the in-memory can be easily overlooked and use significant resources in the cluster.

File: controllers/tempo/tempomonolithic_controller.go [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+		}
+
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, nil
+	}
+
+	// apply defaults
+	tempo.Default()
```
Comment (pavolloffay) [inline controllers/tempo/tempomonolithic_controller.go:59]: Is the default webhook registered as well?

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 7 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:52]: If you're only concerned about OOM situations, the memory counts towards the container resource limit:
> While tmpfs is very fast be aware that, unlike disks, files you write count against the memory limit of the container that wrote them

https://kubernetes.io/docs/concepts/storage/volumes/#emptydir

I'd prefer to keep memory as default, as it's great for quick testing/demos/showcases (and is the default for jaeger all-in-one), and changing it to `pv` is easy (and can/should be mentioned in the docs).

File: controllers/tempo/tempomonolithic_controller.go (hunk 8 above)
Comment (andreasgerstmayr) [reply controllers/tempo/tempomonolithic_controller.go:59]: No, only the validating webhook is registered.
I'm already running with `make run` locally, much faster feedback cycle :)

File: .chloggen/monolithic_mode.yaml (hunk 6 above)
Comment (andreasgerstmayr) [reply .chloggen/monolithic_mode.yaml]: Good idea, I've updated the entry.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 1 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:205]: That would be inconsistent with the
```
<feature>:
  enabled: true
```
style of the CR. I'd prefer to migrate the `TempoStack`, maybe in the next CRD version? Shouldn't be too difficult to create a conversion webhook for this.

```
spec:                                    # TempoMonolithicSpec defines the desired state of TempoMonolithic.
  observability:                         # Observability defines observability configuration for the Tempo deployment
    metrics:                             # Metrics defines the metrics configuration of the Tempo deployment
      prometheusRules:                   # ServiceMonitors defines the PrometheusRule configuration
        enabled: false                   # Enabled defines if the operator should create PrometheusRules for this Tempo deployment
      serviceMonitors:                   # ServiceMonitors defines the ServiceMonitor configuration
        enabled: false                   # Enabled defines if the operator should create ServiceMonitors for this Tempo deployment
```

Comment (pavolloffay) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:205]: so we agreed on using the enabled field which is better supported in tools like `kustomize` and `kubectl edit` (the empty structs are removed). Are we going to reuse some parts from the monolithic APIs?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+
+	// ManagementState defines whether this instance is managed by the operator or self-managed
+	//
+	// +kubebuilder:validation:Optional
+	Management ManagementStateType `json:"management,omitempty"`
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:34]: Asking again, can we reuse some of the structs from the microservies type?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// CA defines the name of a secret containing the CA certificate
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	CA string `json:"ca"`
+
+	// Cert defines the name of a secret containing the TLS certificate and private key
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:155]: what are the entries in the secret?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+type MonolithicJaegerUISpec struct {
+	// Enabled defines if the Jaeger UI should be enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// Ingress defines the ingress configuration for Jaeger UI
+	//
+	// +kubebuilder:validation:Optional
+	Ingress *MonolithicJaegerUIIngressSpec `json:"ingress,omitempty"`
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:172]: Can be the ingress spec reused from the microservices? There are more settings users might want to configure

File: controllers/tempo/tempomonolithic_controller.go (hunk 8 above)
Comment (pavolloffay) [reply controllers/tempo/tempomonolithic_controller.go:59]: We should document it on the defaulting webhook

File: internal/manifests/mutate.go [hunk 12]
```diff
@@ -12,10 +13,22 @@ import (
...
 
+// ImmutableErr occurs if an immutable field should be changed.
+type ImmutableErr struct {
+	field    string
+	existing interface{}
+	desired  interface{}
+}
+
+func (m *ImmutableErr) Error() string {
+	return fmt.Sprintf("update to immutable field %s is forbidden", m.field)
```
Comment (pavolloffay) [inline internal/manifests/mutate.go:29]: shouldn't it print the existing and desired as well?

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 10 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go]: Sorry, I thought I removed this struct already. I'll use a generic TLS struct from TempoStack in the next PR where I add TLS for ingestion.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 1 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:180]: Yes, we can reuse a TLS struct, the multitenancy structs, ManagementStateType, LimitSpec and the storage secret.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 9 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:34]: We can, question is do we value _consistency inside the same CR_ or _consistency between the two CRs_ more?

```
spec:
  observability:                         # Observability defines observability configuration for the Tempo deployment
    metrics:                             # Metrics defines the metrics configuration of the Tempo deployment
      prometheusRules:                   # ServiceMonitors defines the PrometheusRule configuration
        enabled: false                   # Enabled defines if the operator should create PrometheusRules for this Tempo deployment
      serviceMonitors:                   # ServiceMonitors defines the ServiceMonitor configuration
        enabled: false
```

vs

```
spec:
  observability:                         # ObservabilitySpec defines how telemetry data gets handled.
    grafana:                             # Grafana defines the Grafana configuration for operands.
      createDatasource: false            # CreateDatasource specifies if a Grafana Datasource should be created for Tempo.
      instanceSelector:                  # InstanceSelector specifies the Grafana instance where the datasource should be created.
    metrics:                             # Metrics defines the metrics configuration for operands.
      createPrometheusRules: false       # CreatePrometheusRules specifies if Prometheus rules for alerts should be created for Tempo components.
      createServiceMonitors: false       # CreateServiceMonitors specifies if ServiceMonitors should be created for Tempo components.
    tracing:                             # Tracing defines a config for operands.
      jaeger_agent_endpoint: "localhost:6831" # JaegerAgentEndpoint defines the jaeger endpoint data gets send to.
      sampling_fraction: ""
```

Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:34]: The first example is consistent with the rest of the Monolithic CR, and allows additional settings for prometheusRules or serviceMonitors in the future.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 11 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:147]: Already ready in #755:
```
spec:
  jaegerui:                              # JaegerUI defines the Jaeger UI configuration
    enabled: false                       # Enabled defines if the Jaeger UI should be enabled
    ingress:                             # Ingress defines the ingress configuration for Jaeger UI
      annotations:                       # Annotations defines the annotations of the Ingress object.
        "key": ""
      enabled: false                     # Enabled defines if an Ingress object should be created for Jaeger UI
      host: ""                           # Host defines the hostname of the Ingress object.
      ingressClassName: ""               # IngressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource.
    route:                               # Route defines the route configuration for Jaeger UI
      annotations:                       # Annotations defines the annotations of the Ingress object.
        "key": ""
      enabled: false                     # Enabled defines if a Route object should be created for Jaeger UI
      host: ""                           # Host defines the hostname of the Ingress object.
      ingressClassName: ""               # IngressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource.
      termination: "edge"                # Termination specifies the termination type. Default: edge.
```

vs spec of TempoStack:
```
      jaegerQuery:                       # JaegerQuerySpec defines Jaeger Query specific options.
        enabled: false                   # Enabled is used to define if Jaeger Query component should be created.
        ingress:                         # Ingress defines Jaeger Query Ingress options.
          annotations:                   # Annotations defines the annotations of the Ingress object.
            "key": ""
          host: ""                       # Host defines the hostname of the Ingress object.
          ingressClassName: ""           # IngressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource.
          route:                         # Route defines OpenShift Route specific options.
            termination: ""              # Termination specifies the termination type. By default "edge" is used.
          type: ""                       # Type defines the type of Ingress for the Jaeger Query UI. Currently ingress, route and none are supported.
```
(ingressClassName should have been under ingress, as it doesn't apply to route)

Do we value we value _consistency inside the same CR_ or _consistency between the two CRs more_?

File: internal/manifests/mutate.go (hunk 12 above)
Comment (andreasgerstmayr) [reply internal/manifests/mutate.go:29]: I did initially, but printing structs with `fmt.Sprintf("%v", some_struct)` is an unreadable mess if the struct is big.

File: controllers/tempo/tempomonolithic_controller.go (hunk 8 above)
Comment (andreasgerstmayr) [reply controllers/tempo/tempomonolithic_controller.go:59]: Good idea, done.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 9 above)
Comment (rubenvp8510) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:34]: I would prefer consistency on the same CRD, just one question, in the future is possible to have some sort of consolidation in order to get both? That will imply a breaking change though

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
+
+	// ExtraConfig defines any extra (overlay) configuration for components
+	//
+	// +kubebuilder:validation:Optional
+	ExtraConfig *MonolithicExtraConfigSpec `json:"extraConfig,omitempty"`
```
Comment (rubenvp8510) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:39]: Any reason for not reusing the same as microservices? The only reason I think is because the microservices could include other configs in the future. If that is the reason I'm ok

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+}
+
+// MonolithicStorageSpec defines the storage for the Tempo deployment.
+type MonolithicStorageSpec struct {
```
Comment (rubenvp8510) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:43]: One question, Why this have it's own spec and inside only have one structure? Is this for mimic the tempo configuration?

File: apis/tempo/v1alpha1/tempomonolithic_types.go [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicIngestionSpec defines the ingestion settings.
+type MonolithicIngestionSpec struct {
+	// OTLP defines the ingestion configuration for OTLP
+	//
+	// +kubebuilder:validation:Optional
+	OTLP *MonolithicIngestionOTLPSpec `json:"otlp,omitempty"`
```
Comment (rubenvp8510) [inline apis/tempo/v1alpha1/tempomonolithic_types.go:104]: It will be the only protocol supported?

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 11 above)
Comment (rubenvp8510) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:147]: Same comment, I'd prefer consistency on the same CR

File: internal/manifests/mutate.go (hunk 12 above)
Comment (pavolloffay) [reply internal/manifests/mutate.go:29]: shall we then remove fields that are not used in the error struct?

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 14 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:43]: Yes, it's to mimic the tempo configuration. I thought maybe tempo has plans to store other things in the future, so I'll keep the same here also.

But I don't have very strong opinions on this, I could remove that extra layer if you like.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 15 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:104]: I want to make the gateway a drop-in feature, so the service ports should not change if gateway is enabled or not. So I can only support protocols which the gateway also supports.
Afaics the gateway only supports otlp/grpc and otlp/http, right?

With the general move to the OTEL SDK, and using the OTEL collector, I think it's fine to only support OTLP.

File: internal/manifests/mutate.go (hunk 12 above)
Comment (andreasgerstmayr) [reply internal/manifests/mutate.go]: I updated the message to show the result of `cmp.Diff()` now - it's still a bit unreadable as it's a single line, but when replacing the `\n` and `\t` we see a nice diff in the logs, and we already had a dependency on this library anyway (https://github.com/google/go-cmp).

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 13 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go]: I've updated the PR and reused the struct and logic from the TempoStack now.

File: apis/tempo/v1alpha1/tempomonolithic_types.go (hunk 9 above)
Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempomonolithic_types.go:33]: Yes, I was thinking of maybe doing this change in v1alpha2 of TempoStack if we have a consensus.

**Review by pavolloffay — APPROVED** [review]
